src/lib/structures/Argument.ts

```diff
@@ -133,9 +134,9 @@ export interface ArgumentOptions extends AliasPiece.Options {}
 export interface ArgumentContext<T = unknown> extends Record<PropertyKey, unknown> {
 	argument: IArgument<T>;
 	args: Args;
-	message: Message;
-	command: MessageCommand;
-	commandContext: MessageCommand.RunContext;
+	messageOrInteraction: Message | AnyInteraction;
+	command: Command;
+	commandContext: Record<PropertyKey, unknown>;
```

Comment:
hmm, this I don't like, we lose strict typings here... Let's mark command as MessageCommand | ChatInputCommand (or w/e the type is called), and the context as a merge of their contexts

Reply:
Could you explain what you mean by a "merge of their contexts"?

Reply:
ChatInputCommand.Context & MessageCommand.Context maybe

Reply:
I think those two types are the same, so I changed it to `ChatInputCommand.RunContext & MessageCommand.RunContext`.